### ISSUE TITLE
With more than ~150 containers, workers become unresponsive due to volume size checking

### ISSUE THREAD

**Issue by jchesterpivotal** [concourse/concourse#682]
Over the past few days, buildpacks team have observed that our `c4.2xl` worker instances became highly unresponsive. These machines have 8 VCPUs and 16GiB of RAM, and are backed by 15k provisioned-IOP EBS volumes.

So we were a bit surprised when we saw load averages spiking regularly into the 30s when our pipelines were idle. 

We verified that we hadn't gone past the 250 container limit, but our workers were all operating at or under the 200 mark. Most of these are `check` containers.

Further investigation showed that iowait is low or zero. Instead, the CPU is totally consumed by dozens of copies of `/var/vcap/packages/btrfs_tools/sbin/btrfs qgroup show -F --raw /var/vcap/data/baggageclaim/volumes/live/{uuid}/volume`.

In turn we tracked this back to the [GetVolumeSizeInBytes function](https://github.com/concourse/baggageclaim/blob/ce246ab48484f3fbc8527a6c8fd193dada53aa01/volume/driver/btrfs.go#L91) used in the baggageclaim btrfs volume driver.

We understand that the [upcoming rearchitecture of container lifecycles](https://github.com/concourse/concourse/issues/629) should resolve this issue.

In the meantime, our intention is to ameliorate the condition by switching from few large, powerful, fast-disk (and very expensive) workers down to a larger fleet of small, modest workers. By bringing down our volumes per-worker, we think we will avoid the very high load spikes we observed.

**Comment by powerman** [comment]
Is there any temporary workaround for this issue? We've just a couple of concurrent builds but it takes up to 20 minutes before actual work begins. :disappointed:

**Comment by jchesterpivotal** [comment]
Hi @powerman,

Buildpacks team's workaround was to replace 3 large instances with 6 smaller ones. The number of CPUs and amount of RAM was held about constant, but we noticed a big improvement.

We have also upgraded our BOSH stemcell to the latest. Some of the degradation seems to come from Linux kernel version 3.19. Kernel 4.4, shipped in the latest stemcells, has a much later version of btrfs and apparently this improves the performance. We haven't noticed a difference, though we'd already made the switch to 6 workers when we upgraded.
